doc/plugin.rst

```diff
@@ -0,0 +1,108 @@
+##############
+Spyder plugins
+##############
+
+Spyder is made up of multiple plugins that perform a specific task, 
+for instance spyder-terminal or the variable explorer. As well, plugins are
+able to communicate among others and with spyder. In this way, 
+some of the new functionalities for Spyder require a brand new plugin. 
+
+============================
+Creating a new spyder plugin
+============================
+
+So, you want to build a spyder plugin but you are not sure of how to start?
+This documentation will guide you throghout the process.
+
+Please before starting the tutorial, make sure you have the following dependencies installed:
+
+- Git
+- Spyder
+- Python
+- Anaconda
+
+
+Getting started
+---------------
+All spyder plugins have the same structure and it is important for the new ones to follow the same set of rules.
+We developed a template that all the plugins must follow that is available as a `cookiecutter <https://github.com/spyder-ide/spyder-plugin-cookiecutter>`_.
+
+1. Clone or download `spyder-pugin-cookiecutter <https://github.com/spyder-ide/spyder-plugin-cookiecutter>`_
+2. Install cookiecutter with conda
+
+.. code-block:: bash
+
+   conda install -c conda-forge cookiecutter
+
+3. Run against the repo
+
+.. code-block:: bash
+
+   cookiecutter gh:spyder-ide/spyder-plugin-cookiecutter
+
+In this step you will need to input some values for your plugin as shown below.
+Normally, you don't need to change *project_name* and *repo_name*.
+
+.. image:: images/plugin/create-example-plugin.png
```

Comment:
Just leaving a note here to mention that this image needs to change to remove some options not needed from our cookiecutter template.